lib/models/addon.js

```diff
@@ -331,7 +334,29 @@ let addonProto = {
   */
   isDevelopingAddon() {
     if (process.env.EMBER_ADDON_ENV === 'development' && (this.parent instanceof Project)) {
-      return this.parent.name() === this.name;
+      const parentName = this.parent.name();
+      // If the name in package.json and index.js match, we're definitely developing
+      if (parentName === this.name) {
+        return true;
+      }
+
+      // If package.json and index.js names do not match, we should check for scoped vs non-scoped
+      const parsed = npa(parentName);
+      let parsedName = parsed.name;
+      if (parsed.scope) {
+        parsedName = parsedName.replace(parsed.scope, '').slice(1);
+      }
+      if (parsedName === this.name) {
+        if (!alreadyWarnedAddonNames.includes(this.name)) {
```

Comment:
I'm wondering if we need this check. The warning should only be emitted for addon authors, and only for their specific addon.  In most cases I would assume it would only be emitted once (because its somewhat odd to have multiple versions of the same addon in the addons own build), and even in the less likely chance that it does get emitted twice it seems not _too_ bad...

My concern with this sort of module scoped guard is that we have no way to reset it during the lifetime of the running node process. This will make testing difficult, and in some cases will hide the warning when it _should_ be emitted...

Reply:
So do you think we should remove the `includes` check here? I was getting it logged 8 times running in my addon, which could get noisy.

Reply:
Ah, we probably check `isDevelopingAddon` many times for the same instance.

Lets remove the module scope array and keep a local `this._hasWarnedForMismatchedNames = true` that we use instead.  What do you think?

Reply:
That sounds fine. Will push up a change that does that.

Reply:
Changed to use your suggestion. Let me know if anything else needs tweaking.